File: c/raindrops/raindrops.c
#include "raindrops.h"
#include <assert.h>
#include <string.h>
#include <stdio.h>

static char VALUES[] = { 3, 5, 7 };
static const char *SOUNDS[] = { "Pling", "Plang", "Plong" };

char *convert(char result[], int drops){
  assert(result);

  for (size_t i = 0; i < (sizeof(VALUES) / sizeof(VALUES[0])); i++)
    if (!(drops % VALUES[i])) strcat(result, SOUNDS[i]);

  if (!*result) sprintf(result, "%i", drops);

  return result;
}

File: javascript/raindrops/raindrops.js
const LOOKUP = {
  3: "Pling",
  5: "Plang",
  7: "Plong"
}

export const convert = (number) =>
  Object.entries(LOOKUP).reduce((noise, [factor, sound]) => 
    noise + (number % factor ? '': sound), '') || number.toString();
